src/core/components/ping-pull-stream.js

```diff
@@ -27,7 +27,8 @@ module.exports = function pingPullStream (self) {
       if (err) {
         log.error(err)
         source.push(getPacket({ success: false, text: err.toString() }))
-        source.end(err)
+        source.push(err)
+        source.end()
```

Comment:
This made the tests pass but I'm not 100% convinced this is exactly what we want.

Reply:
We need to differentiate between a real failure (invalid peer ID) and unavailable.

Reply:
I've separated the getPeer from the runPing steps